src/lib/rng/system_rng/system_rng.cpp

```diff
@@ -158,10 +165,16 @@ class System_RNG_Impl final : public RandomNumberGenerator {
       std::string name() const override { return "CCRandomGenerateBytes"; }
 
    private:
+#ifdef BOTAN_HAS_JITTER
+      rand_data *m_jitter;
+#endif
       void fill_bytes_with_input(std::span<uint8_t> output, std::span<const uint8_t> /* ignored */) override {
          if(::CCRandomGenerateBytes(output.data(), output.size()) != kCCSuccess) {
             throw System_Error("System_RNG CCRandomGenerateBytes failed", errno);
          }
+#ifdef BOTAN_HAS_JITTER
+         jitter_buffer(m_jitter, output);
```

Comment:
Categorically we do not do this without end-user (ie in application source code) consent. If an application developer types `System_RNG` that means they trust, specifically, the RNG provided by the OS vendor. They are not buying into "The System_RNG XORed with something else that I may be unaware of".

The developer is allowed to type `Jitter_RNG` and buy into "yeah lets use this thing", and then it is visible at the source code level what is happening.